baybe/surrogates/gaussian_process/presets/default.py

```diff
@@ -22,49 +22,44 @@
 class DefaultKernelFactory(KernelFactory):
     """A factory providing the default kernel for Gaussian process surrogates.
 
-    The logic is adapted from EDBO (Experimental Design via Bayesian Optimization).
-
-    References:
-    *   https://github.com/b-shields/edbo
-    *   https://doi.org/10.1038/s41586-021-03213-y
+    This is taking the low and high dimensional limits of
+    :class:`baybe.surrogates.gaussian_process.presets.default.EDBOKernelFactory`
+    and interpolates the prior moments linearly between them.
     """
 
     def __call__(  # noqa: D102
         self, searchspace: SearchSpace, train_x: Tensor, train_y: Tensor
     ) -> Kernel:
         # See base class.
+        effective_dims = searchspace.n_effective_default_kernel_dimensions
 
-        mordred = (searchspace.contains_mordred or searchspace.contains_rdkit) and (
-            train_x.shape[-1] >= 50
-        )
+        # Fix dimensionality limits
+        LOW_D_LIM, HIGH_D_LIM = 8, 100
 
-        # low D priors
-        if train_x.shape[-1] < 10:  # <-- different condition compared to EDBO
+        # Apply prior logic
+        if effective_dims < LOW_D_LIM:
+            # Low D priors
             lengthscale_prior = GammaPrior(1.2, 1.1)
             lengthscale_initial_value = 0.2
             outputscale_prior = GammaPrior(5.0, 0.5)
             outputscale_initial_value = 8.0
+        elif effective_dims > HIGH_D_LIM:
+            # High D priors
+            lengthscale_prior = GammaPrior(2.5, 0.55)
+            lengthscale_initial_value = 6.0
+            outputscale_prior = GammaPrior(3.5, 0.15)
+            outputscale_initial_value = 15.0
+        else:
+            # Interpolate prior moments linearly between low D and high D regime
+            interp_factor = (effective_dims - LOW_D_LIM) / (HIGH_D_LIM - LOW_D_LIM)
 
-        # DFT optimized priors
-        elif mordred and train_x.shape[-1] < 100:
-            lengthscale_prior = GammaPrior(2.0, 0.2)
-            lengthscale_initial_value = 5.0
-            outputscale_prior = GammaPrior(5.0, 0.5)
-            outputscale_initial_value = 8.0
-
-        # Mordred optimized priors
-        elif mordred:
-            lengthscale_prior = GammaPrior(2.0, 0.1)
-            lengthscale_initial_value = 10.0
-            outputscale_prior = GammaPrior(2.0, 0.1)
-            outputscale_initial_value = 10.0
+            def _interp(a, b):
```

Comment:
I think you are reinventing the wheel here. Why not simply take `numpy.interp`? If you further pull out the prior float numbers first and interpolate them before creation the `GammaPrior` objects in our if-else blocks, you can even reuse a single `GammaPrior` call for each component, since `numpy.interp` by default keeps the values constant outside the specified boundaries. I think the final code will then have like 30% of its current length + no need to define custom interpolators.

Reply:
forgot this exists, added it